### PR TITLE
[BUG_FIX] fix check for agg rules in detector trigger condition to create chained findings monitor

### PR DESCRIPTION
**Pull request by eirsep**
### Description
remove check for agg rules in detector trigger condition to create chained findings monitor.
Currently, chained findings monitor is not created unless ids of aggregation Sigma rules are not mentioned in the detector trigger condition
Right behaviour is to create chained findings monitor irrespective of trigger condition.
Add agg rule tags to chained findings monitor
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by jowg-amazon** [comment]
Without this additional check in the logic won't we call `createDocLevelMonitorMatchAllRequest` even for non aggregation rules? Is this the expected behavior?

**Comment by eirsep** [comment]
> Without this additional check in the logic won't we call `createDocLevelMonitorMatchAllRequest` even for non aggregation rules? Is this the expected behavior?

fixed to check for atleast one agg rule

File: src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java [hunk 1]
```diff
@@ -828,20 +827,30 @@ private void addThreatIntelBasedDocLevelQueries(Detector detector, ActionListene
      */
     private IndexMonitorRequest createDocLevelMonitorMatchAllRequest(
             Detector detector,
-            WriteRequest.RefreshPolicy refreshPolicy,
+            RefreshPolicy refreshPolicy,
             String monitorId,
-            RestRequest.Method restMethod
-    ) {
+            Method restMethod,
+            List<Pair<String, Rule>> queries) {
```
Comment (jowg-amazon) [inline src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java:833]: If there are multiple aggregation rules/queries, do the queried passed in here include all of them or only the aggregation rule that created the bucket level monitor?

Comment (eirsep) [reply src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java:834]: we are only adding the tags of each aggregation query to the doc level monitor's DocLevelQuery tags. Btw each agg rule creates 1 bucket level monitor. For all bucket level monitors there is one chained findings doc level monitor

**Review by sbcd90 — APPROVED** [review]

**Review by jowg-amazon — APPROVED** [review]

**Review by sbcd90 — APPROVED** [review]

**Review by AWSHurneyt — APPROVED** [review]

**Comment by eirsep** [comment]
Flaky tests failed
```
- org.opensearch.securityanalytics.correlation.CorrelationEngineRestApiIT.testBasicCorrelationEngineWorkflowWithRolloverByMaxDoc
 - org.opensearch.securityanalytics.resthandler.DetectorMonitorRestApiIT.testRemoveDocLevelRuleAddAggregationRules_verifyFindings_success
 - org.opensearch.securityanalytics.resthandler.DetectorMonitorRestApiIT.testRemoveDocLevelRuleAddAggregationRules_verifyFindings_success
 - org.opensearch.securityanalytics.threatIntel.integTests.ThreatIntelJobRunnerIT.testCreateDetector_threatIntelEnabled_testJobRunner

**Pull request merged**